File: Tai_lieu/Code/palinless.cpp
#include <bits/stdc++.h>
using namespace std;

bool dx(string s,int l, int r)
{
    while (l<r)
    {
        if (s[l]!=s[r]) return 0;
        l++;r--;
    }
    return 1;
}
string check(string s)
{
    int n=s.size();
    if (n<3) return "YES";
    return ((dx(s,0,n-1)||dx(s,1,n-1))? "YES" : "NO" );
}


int main()
{
    ios_base::sync_with_stdio(false);cin.tie(0);cout.tie(0);
    int t; cin >> t;
    while (t--)
    {
        string s;
        cin >> s;
        cout << check(s) << endl;
    }
}

File: Tai_lieu/Code/palinless.py
t = int(input())
for _ in range(t):
	s = input()
	s = s + s[0]
	l,r = 0,len(s) - 1
	while l<=r:
		if (s[l] == s[r]):
			r-=1
			l+=1
		else:
			break
	if (r < l):
		print("YES")
	else:
		print("NO")